Plugins: update notices to the new format

johnHackworth opened the pull request:
This PR updates notices in /plugins/ to the new format introduced in https://github.com/Automattic/wp-calypso/pull/1603/files

It both updates the 'update' flags inside of each plugin item

Before:
![image](https://cloud.githubusercontent.com/assets/1554855/11845782/b7982044-a415-11e5-903b-8e405918ab69.png)

After:
![image](https://cloud.githubusercontent.com/assets/1554855/11845731/8478740c-a415-11e5-92e9-e938a54156a0.png)


And the full-width notice error messages:
Before:
![image](https://cloud.githubusercontent.com/assets/1554855/11845792/c69fecac-a415-11e5-907d-3bb3599beeb0.png)

After:
![image](https://cloud.githubusercontent.com/assets/1554855/11845765/a69cd0fa-a415-11e5-88d4-c261adf8d5c1.png)


How to test
========
1. Go to http://calypso.dev:3000/plugins/
2. Check that any update flag you have in the plugin list is rendered with the new notice-compact format
3. Click on "bulk edit"
4. Select a plugin and do any action that would result in a error (you probably have to have a failing jetpack site for this);
5. Check the error notice is shown full-width

mtias commented on client/my-sites/plugins/plugin-item/plugin-item.jsx:
```diff
@@ -98,12 +98,11 @@ module.exports = React.createClass( {
-					</span>
-				</div>
+				<Notice isCompact
+					showDismiss={ false }
```
this should be assumed for `isCompact`

mtias commented on client/my-sites/plugins/plugin-item/style.scss:
```diff
@@ -82,15 +91,15 @@
 	font-size: 14px;
 	font-weight: 600;
-	white-space: pre;
+	margin-top: 0px;
```
do you need the margin now?

mtias commented:
Looks good!

mtias commented on client/my-sites/plugins/plugin-item/plugin-item.jsx:
```diff
@@ -14,7 +14,7 @@ var CompactCard = require( 'components/card/compact' ),
 	PluginActivateToggle = require( 'my-sites/plugins/plugin-activate-toggle' ),
 	PluginAutoupdateToggle = require( 'my-sites/plugins/plugin-autoupdate-toggle' ),
-	Notice = require( 'notices/notice' ),
+	Notice = require( 'components/notice' ),
```
\o/

rickybanister commented:
I think we should not set the bottom margin of the regular sized notice to `0`. When it pushes down the next card it makes it obvious that it is attached to the top one.

johnHackworth commented:
@rickybanister do you mean leaving it like this?

![image](https://cloud.githubusercontent.com/assets/1554855/11846964/68a3a1ba-a41b-11e5-84f9-5a925bc1a4a7.png)

johnHackworth commented:
I have removed the margin-bottom: 0 to see how it works. Also, now all the small notifications under the plugin name uses a `notice compact`, so we can discard a lot of custom css and avoid size jump when the contents in that area changes:

![image](https://cloud.githubusercontent.com/assets/1554855/11847228/b33c2f34-a41c-11e5-85d2-4fe57da02ed6.png)

In the above screencap, 'A newer version ... ', 'Last updated...', and 'Deactivatig' are all '<Notice compact>'

johnHackworth commented:
(also, I adjusted the notice-compact height (to the values they will have as default when https://github.com/Automattic/wp-calypso/pull/1676 is merged) & the line-height of the plugin name to make them fit into the plugin icon height)

mtias commented on client/my-sites/plugins/plugin-item/style.scss:
```diff
@@ -7,8 +7,30 @@
+		}
+	}
+
+	.notice.is-in-progress,
```
I think these should use some of the default statuses, `is-info`?

rickybanister replied:
I'd agree with that. The standard 'Last updated' text doesn't need a status, but when we are updating or deactivating we could make it more visible.

mtias replied:
Yeah.

johnHackworth commented:
new version!

![image](https://cloud.githubusercontent.com/assets/1554855/11848856/39d7f700-a425-11e5-8454-c8a47156b461.png)

the status flags now use regular notices (no transparent background) and the standing texts there ('last updated ... ') are not notices at all

mtias commented on client/my-sites/plugins/plugin-item/style.scss:
```diff
@@ -4,12 +4,22 @@
-	.plugins-list &.is-compact:last-child {
-		margin-bottom: 0;
+
+	.notice.is-compact {
```
All of this should be removed once #1676 lands.

johnHackworth replied:
done!

rickybanister commented:
@mtias I know we can throw a custom icon in notices, but can we make it have no icon? For the 'Activating' notice, it doesn't seem necessary to show the little `(i)`

mtias commented:
If we think it's something we'll do often, we can add a prop. Otherwise just a `.your-component .notice__icon { display: none; }` :D

johnHackworth commented:
I removed the non-needed notices styles in this PR and removed the icon in the is-info notices inside of plugin-item:

![image](https://cloud.githubusercontent.com/assets/1554855/11852963/6ac89c50-a43b-11e5-8461-bc6332c347f0.png)

rickybanister commented:
Perfect, looking really sharp. Ship ship ship

enejb commented:
I wasn't able to see any of the inline error notices any more. Were they removed?

johnHackworth commented:
@enejb they are shown only if you are in bulk edit mode, for some reason ... in this PR https://github.com/Automattic/wp-calypso/pull/1554 I'm changing it to be shown in both modes

enejb commented:
I just pushed a rebase here. I think this is ready to merge!
